app/controllers/api/comments_controller.rb

```diff
@@ -0,0 +1,39 @@
+module Api
+  class CommentsController < ApplicationController
+    skip_before_action :authenticate_user!, only: %i[index create]
+    skip_before_action :verify_authenticity_token
+    # before_action :authenticate_user!, only: [:create]
+    before_action :set_user_post
+
+    def index
+      @user = User.find(params[:user_id])
+      @post = @user.posts.find(params[:post_id])
+      @comments = @post.comments
+      render json: @comments
+    end
+
+    def create
+      @user = User.find(params[:user_id])
+      @post = @user.posts.find(params[:post_id])
+      @comment = @post.comments.new(comment_params)
+      @comment.user = @user
+
+      if @comment.save
+        render json: @comment, status: :created
+      else
+        render json: { error: 'Comment not created' }, status: :unprocessable_entity
+      end
+    end
```

Comment:
- Good job done so far, but I am still unable to create a comment in or a  logged in user, please make sure the API authentication works on as  expected 🙏